lib/analyzer/docker-analyzer.ts

```diff
@@ -19,6 +19,18 @@ export async function analyze(
   dockerfileAnalysis?: dockerFile.DockerFileAnalysis,
   options?: DockerOptions,
 ) {
+  try {
+    await imageInspector.pullIfNotLocal(targetImage, options);
+  } catch (error) {
+    if (error.stderr.includes("pull access denied")) {
+      throw new Error(
+        "Docker error: image was not found locally and pulling failed: " +
```

Comment:
why does the `pull access denied` error gets "special treatment" and you actually convert the error to this string, and the other cases you throw as is? I assume that either you always throw the error as-is or throw this "Docker error" instead